iam_robolib/src/feedback_controller/cartesian_impedance_feedback_controller.cpp

```
@@ -101,8 +104,14 @@ void CartesianImpedanceFeedbackController::get_next_step(const franka::RobotStat
   Eigen::Vector3d position(transform.translation());
   Eigen::Quaterniond orientation(transform.linear());
 
-  Eigen::Vector3d position_d(traj_generator->desired_position_);
-  Eigen::Quaterniond orientation_d(traj_generator->desired_orientation_);
+  PoseTrajectoryGenerator* pose_trajectory_generator = dynamic_cast<PoseTrajectoryGenerator*>(traj_generator);
+
+  if(pose_trajectory_generator == nullptr) {
+    throw 333;
```

what is this? Let's raise a correct exception here.